README.md

```diff
@@ -0,0 +1,31 @@
+# Ubuntu Keyboard
+
+This is a QML and C++ based Keyboard Plugin for Maliit, based on the Maliit Reference plugin, taking into account the special UI/UX requests of Ubuntu Phone.
+
+# Updating presage language models
+
+If the source material used for presage's prediction models is update the databases can be regenerated by running "make lang_db_$lc" within the plugin's "src" directory (where $lc is the language code for the updated language, e.g. "make lang_db_en" for English)
+
+# Updating translations
+
+After building the keyboard (such as with `dpkg-buildpackage -us -uc`), run `make pot` from the `po` directory to rebuild the translation source file.
+
+# On-device testing
+
+To test a pull request on your device, type
+
+    sudo ubports-qa install PR-X
+
+where `X` is the code of the request.
```

Comment:
>The syntax is supposed to be `ubports-qa install keyboard-component X`

Fix'd.

> but there are bugs in ubports-qa currently

Frankly I would like to build the package locally in a distro agnostic way (I see the `snapcraft.yml` in the directory but `snapcraft` does not produce anything when executed) but actually the CI you set up is really nice.

I will now try to build the package regularly with the given comand in a VM.

(OT: can I dist-upgrade libertine containers to disco, or are we stuck on xenial in containers, too?)

Reply:
I tried building on both Ubuntu xenial and Ubuntu disco with `dpkg-buildpackage -us -uc` but it [fails](https://paste.ubuntu.com/p/pW3JhPbdwn/).

Trying to find why `eventdispatcher` is unknown, I stumbled upon [ubuntu-ui-toolkit](https://code.launchpad.net/ubuntu/+source/ubuntu-ui-toolkit), which it fails to build on [xenial](https://paste.ubuntu.com/p/MChkrTyWwX/) and disco (because missing packages. Maybe ubports has a x86_64 binary/source repo I have to attach to?).

Then I found your [fork](https://github.com/ubports/ubuntu-ui-toolkit), which has broken dependencies on both versions.

Reply:
I tried building on device following [docs](http://docs.ubports.com/en/latest/systemdev/testing-locally.html) but of course I run out of space installing dependecies.

Why development mode does not move the actual system on the `userdata` partition?

Reply:
Bear in mind that building on your device will result in a high amount of writes to your flash storage, reducing the lifetime potentially much faster than what you want.
The reason why everything is locked down is besides making OTA updates possible also to reduce wear of the flash which is crucial for longevity.
So we do not encourage building on devices, also because of that reason.

Reply:
@Flohack74 Thank you for that, I forgot it.

So, how exactly one does compile this? Is there a repo I have to add to 16.04 to get to compile it?